### DISCUSSION TITLE
Help getting streamlink CLI output trough Python subprocess

### DISCUSSION THREAD

**Discussion by NeoFahrenheit** [streamlink/streamlink#4878] [Q&A] [answered]
So, I posted about this question here: [Stack Overflow question](https://stackoverflow.com/questions/74002131/reading-the-countinous-output-of-a-streamlink-cli-with-subprocess-popen)

Basically, I'm doing a GUI that manages and records streams. As I would like to capture the download information from streamlink, I have to capture it's output on the terminal / command prompt, which is done by erasing the current line and replacing it with another (\r, I suposse). Through the subprocess, I can get every output line, except for that one.

Any help is appreciated. :D

**Comment by bastimeyer** [comment, marked as answer]
This is a general programming question and doesn't belong here. But let me give you at least a bit of a useful reply...

Why do you intent on reading `streamlink_cli`'s output instead of properly integrating `streamlink`'s APIs if you're already using Python for building a GUI? That doesn't make any sense. Streamlink's CLI is not actually meant for that and you'll face lots of issues. When integrating Streamlink's Python API, you don't have to rely on any restrictive CLI stuff and its interfaces that are considered unstable, like its output format for example.

And when running a subprocess in Python, you should never pass a command line string for execution, because that has to be parsed and tokenized and thus needs to be properly sanitized when using random user inputs, which will most likely not work as you expect.
https://docs.python.org/3/library/subprocess.html#subprocess.Popen

Your main issue is that `streamlink_cli`'s progress output is written to `stderr`, not `stdout`, hence why you don't see anything in your script's output.

I'm not going to comment on any of the other questions. You'll have to figure this out on your own. Good luck.
